packages/schemas/tables/user_logs.sql

```diff
@@ -0,0 +1,13 @@
+create type log_type as enum ('SignInUsernameAndPassword', 'ExchangeAccessToken');
```

Comment:
1. `SignInUsernameAndPassword` may be too specific? will it be better to split into `user_log_event` and `user_log_type`?
2. let's add `user_log_` prefix to keep the naming convention for custom types, unless it's a general type. (`password_encryption_method` in `packages/schemas/tables/users.sql` needs to be updated, created [LOG-365](https://linear.app/silverhand/issue/LOG-356/add-prefix-for-password-encryption-method))

@simeng-li any thoughts on this?

Reply:
We won't have too many enum values of `user_log_type`, if need to split into subsets in the future, we can bring in a second layer in UI or analytic service. Kind of over engineering for now.

Reply:
agreed, if so, we could replace the `log_result` with `log_event`. 
 
 Couple of further open questions: 
 1. Are we going to have other customized log tables besides current `user_logs`?
 2. if so, should we align the log schema and naming conventions,
 3. if we keep a general log schema shared cross the platform then could use `log_type` + `log_event` + `data` to identify a specific log circumstance.  
 
For the log data schema it self:

i.e. 
```
log_type: 'log_type_api',
log_event: 'log_event_authentication',  // could be more specific
event_data:  {
  api: '/auth?code=XXXX',
  status: 404,
  response: {  error_message: '' }
}
```

```
log_type: `log_type_user`,
log_event: 'log_event_authorization_login',
event_data: {
  userid: XXX,
  clientId: XXX,
  success: false,
  error_msg: '',
}
```

And each `log_type` could have a specific event_data schema defined. 

Implement a general logging method, takes the general log schema as its input.   Based on different `log_type` we may have different table route & data insert logic. based on each one's  `event_data` schema.

Reply:
User-log would be the only one that will be retrived by user directly, so is easier to seperated from other logs considering performance.